### DISCUSSION TITLE
ascii2nc can't open input ASCII file

### DISCUSSION THREAD

**Discussion by JosemariePalero** [dtcenter/METplus#2638] [File I/O] [answered]
Hi! I would like to ask for guidance regarding ASCII2NC tool. I can't figure out how to fix this error.

Here are the details:
ascii2nc \ ${METPLUS_DATA}synopPCP202112010000.ascii \ ${METPLUS_DATA}synopPCP202112010000.nc -v 2
DEBUG 1: Start ascii2nc by erdanx(1000) at 2024-07-16 05:16:53Z  cmd: ascii2nc  /home/erdanx/ai4rp/WRF/practice/formattedPCPSynop/synopPCP202112010000.ascii  /home/erdanx/ai4rp/WRF/practice/formattedPCPSynop/synopPCP202112010000.nc -v 2 
DEBUG 1: Config File: /home/erdanx/WRF/MET-11.0.2/share/met/config/Ascii2NcConfig_default
ERROR  : 
ERROR  : determine_ascii_format() -> can't open input ASCII file " /home/erdanx/ai4rp/WRF/practice/formattedPCPSynop/synopPCP202112010000.ascii" for reading
ERROR  : 

my ASCII file looks like this:
ADPSFC 98223 20211201_000000    18.1831   120.5347 -9999.00  61  24 -9999.00 NA   0.01
ADPSFC 98232 20211201_000000    18.3602   121.6304 -9999.00  61  24 -9999.00 NA   8.00
ADPSFC 98325 20211201_000000    16.0868   120.3521 -9999.00  61  24 -9999.00 NA -9999.00
ADPSFC 98328 20211201_000000    16.4040   120.6015 -9999.00  61  24 -9999.00 NA -9999.00
ADPSFC 98336 20211201_000000    16.2651   122.1288 -9999.00  61  24 -9999.00 NA   1.00
ADPSFC 98429 20211201_000000    14.5047   121.0048 -9999.00  61  24 -9999.00 NA   4.00
ADPSFC 98430 20211201_000000    14.6451   121.0443 -9999.00  61  24 -9999.00 NA -9999.00
ADPSFC 98431 20211201_000000    13.4146   121.1869 -9999.00  61  24 -9999.00 NA -9999.00
ADPSFC 98440 20211201_000000    14.1286   122.9826 -9999.00  61  24 -9999.00 NA  99.50
ADPSFC 98444 20211201_000000    13.1506   123.7284 -9999.00  61  24 -9999.00 NA -9999.00
ADPSFC 98526 20211201_000000    12.0035   120.2000 -9999.00  61  24 -9999.00 NA -9999.00
ADPSFC 98531 20211201_000000    12.3597   121.0479 -9999.00  61  24 -9999.00 NA -9999.00
ADPSFC 98536 20211201_000000    12.5786   122.2703 -9999.00  61  24 -9999.00 NA -9999.00
ADPSFC 98543 20211201_000000    12.3663   123.6292 -9999.00  61  24 -9999.00 NA -9999.00
ADPSFC 98618 20211201_000000     9.7401   118.7586 -9999.00  61  24 -9999.00 NA -9999.00
ADPSFC 98630 20211201_000000    10.8541   121.0082 -9999.00  61  24 -9999.00 NA -9999.00
ADPSFC 98646 20211201_000000    10.3223   123.9801 -9999.00  61  24 -9999.00 NA  99.20
ADPSFC 98653 20211201_000000     9.7828   125.4894 -9999.00  61  24 -9999.00 NA  99.20
ADPSFC 98741 20211201_000000     8.5996   123.3437 -9999.00  61  24 -9999.00 NA  99.60
ADPSFC 98753 20211201_000000     7.1276   125.6550 -9999.00  61  24 -9999.00 NA -9999.00
ADPSFC 98755 20211201_000000     8.3675   126.3385 -9999.00  61  24 -9999.00 NA   0.01
ADPSFC 98836 20211201_000000     6.9171   122.0663 -9999.00  61  24 -9999.00 NA   2.00
ADPSFC 98329 20211201_000000    15.7359   120.9368 -9999.00  61  24 -9999.00 NA -9999.00
ADPSFC 98545 20211201_000000    12.8394   123.9970 -9999.00  61  24 -9999.00 NA -9999.00
ADPSFC 98550 20211201_000000    11.2435   125.0078 -9999.00  61  24 -9999.00 NA   0.01
ADPSFC 98602 20211201_000000    11.0521   114.2850 -9999.00  61  24 -9999.00 NA   0.01
ADPSFC 98747 20211201_000000     8.5357   124.5579 -9999.00  61  24 -9999.00 NA   0.01
ADPSFC 98132 20211201_000000    20.7870   121.8384 -9999.00  61  24 -9999.00 NA   3.00
ADPSFC 98133 20211201_000000    19.2630   121.4670 -9999.00  61  24 -9999.00 NA  21.00
ADPSFC 98222 20211201_000000    17.8901   120.4597 -9999.00  61  24 -9999.00 NA -9999.00
ADPSFC 98233 20211201_000000    17.6477   121.7585 -9999.00  61  24 -9999.00 NA   7.00
ADPSFC 98324 20211201_000000    15.3262   119.9690 -9999.00  61  24 -9999.00 NA -9999.00
ADPSFC 98327 20211201_000000    15.1822   120.5617 -9999.00  61  24 -9999.00 NA   0.01
ADPSFC 98334 20211201_000000    15.7488   121.6320 -9999.00  61  24 -9999.00 NA -9999.00
ADPSFC 98425 20211201_000000    14.5884   120.9679 -9999.00  61  24 -9999.00 NA -9999.00
ADPSFC 98426 20211201_000000    14.7919   120.2706 -9999.00  61  24 -9999.00 NA   7.00
ADPSFC 98427 20211201_000000    14.0184   121.5966 -9999.00  61  24 -9999.00 NA -9999.00
ADPSFC 98428 20211201_000000    14.4950   120.9068 -9999.00  61  24 -9999.00 NA -9999.00
ADPSFC 98432 20211201_000000    14.0877   121.0624 -9999.00  61  24 -9999.00 NA -9999.00
ADPSFC 98433 20211201_000000    14.5812   121.3693 -9999.00  61  24 -9999.00 NA -9999.00
ADPSFC 98434 20211201_000000    14.7483   121.6779 -9999.00  61  24 -9999.00 NA   5.00
ADPSFC 98435 20211201_000000    14.1054   122.0177 -9999.00  61  24 -9999.00 NA   5.00
ADPSFC 98446 20211201_000000    13.5779   124.2079 -9999.00  61  24 -9999.00 NA  99.50
ADPSFC 98538 20211201_000000    11.6002   122.7497 -9999.00  61  24 -9999.00 NA -9999.00
ADPSFC 98546 20211201_000000    12.5054   124.6285 -9999.00  61  24 -9999.00 NA -9999.00
ADPSFC 98548 20211201_000000    11.7750   124.8842 -9999.00  61  24 -9999.00 NA   2.00
ADPSFC 98553 20211201_000000    11.6600   125.4423 -9999.00  61  24 -9999.00 NA -9999.00
ADPSFC 98558 20211201_000000    11.0456   125.7555 -9999.00  61  24 -9999.00 NA -9999.00
ADPSFC 98642 20211201_000000     9.3354   123.3033 -9999.00  61  24 -9999.00 NA -9999.00
ADPSFC 98644 20211201_000000     9.5842   123.8160 -9999.00  61  24 -9999.00 NA  11.00
ADPSFC 98648 20211201_000000    10.1390   124.8604 -9999.00  61  24 -9999.00 NA   2.00
ADPSFC 98746 20211201_000000     7.1617   124.2148 -9999.00  61  24 -9999.00 NA -9999.00
ADPSFC 98751 20211201_000000     8.1514   125.1339 -9999.00  61  24 -9999.00 NA   3.00
ADPSFC 98752 20211201_000000     8.9471   125.4823 -9999.00  61  24 -9999.00 NA   0.01
ADPSFC 98851 20211201_000000     6.0573   125.1031 -9999.00  61  24 -9999.00 NA -9999.00

**Comment by jprestop** [comment, marked as answer]
HI @JosemariePalero.  Thank you for this information and for submitting what your data looks like.  

Which version of MET are you using?  Using MET-11.1.0 with the data you provided, I am unable to reproduce your problem.  

I wonder if there is a permissions issue with the file `/home/erdanx/ai4rp/WRF/practice/formattedPCPSynop/synopPCP202112010000.ascii`.  Please check to ensure that read permissions are open for your user account and let us know how it goes.

```
orion-devel-1[35] jprestop$ ascii2nc test.txt test.nc -v 2
DEBUG 1: Start ascii2nc by jprestop(8132) at 2024-07-16 23:52:23Z  cmd: ascii2nc test.txt test.nc -v 2 
DEBUG 1: Config File: /apps/contrib/MET/11.1.0/share/met/config/Ascii2NcConfig_default
DEBUG 2: Rejected 0 observations off the masking grid.
DEBUG 2: Rejected 0 observations outside the masking polyline.
DEBUG 2: Rejected 0 observations not matched with station ID's.
DEBUG 2: Processing observations for 55 headers.
DEBUG 1: Creating NetCDF Observation file: test.nc
DEBUG 2: Finished processing 55 observations for 55 headers.
DEBUG 1: Finish ascii2nc by jprestop(8132) at 2024-07-16 23:52:24Z

```

**Reply by JosemariePalero** [reply]
Hi. Thank you for your guidance. Updating the MET version to MET-11.1.0 solves the issue for me.

**Reply by jprestop** [reply]
Great news @JosemariePalero!  Thank you for letting us know.  I'll go ahead and mark this discussion as answered and lock it.  Please don't hesitate to open a new discussion with any new questions.
